src/Microsoft.FeatureManagement/Allocation/Allocation.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System.Collections.Generic;
+
+namespace Microsoft.FeatureManagement
+{
+    /// <summary>
+    /// The definition of how variants are allocated for a feature.
+    /// </summary>
+    public class Allocation
+    {
+        /// <summary>
+        /// The default variant used if the feature is enabled and no variant is assigned.
+        /// </summary>
+        public string DefaultWhenEnabled { get; set; }
+
+        /// <summary>
+        /// The default variant used if the feature is disabled.
+        /// </summary>
+        public string DefaultWhenDisabled { get; set; }
+
+        /// <summary>
+        /// Describes a mapping of user ids to variants.
+        /// </summary>
+        public IEnumerable<UserAllocation> User { get; set; }
+
+        /// <summary>
+        /// Describes a mapping of group names to variants.
+        /// </summary>
+        public IEnumerable<GroupAllocation> Group { get; set; }
+
+        /// <summary>
+        /// Allocates percentiles of user base to variants.
+        /// </summary>
+        public IEnumerable<PercentileAllocation> Percentile { get; set; }
+
+        /// <summary>
+        /// Maps users to the same percentile across multiple feature flags.
+        /// </summary>
+        public string Seed { get; set; }
```

Comment:
In our design doc, seed is a number.

Reply:
My fault, changed this while using `Bind` but didn't call it out. No reason for it anymore, I'll change it to `int`.

Reply:
I think a string is actually a good idea. We're appending it as a string anyway in our implementation. I think a string might be better if it's used in multiple places, because customers could use names if they wanted, like "BrowserExperiments".

Reply:
Started a separate thread to discuss whether to keep it as number or move to string.

In the event we stick with number, @amerjusupovic I suppose we can use nullable int?

Reply:
Notable usage of `Seed` pattern in .NET [Random](https://learn.microsoft.com/en-us/dotnet/api/system.random.-ctor?view=net-7.0#system-random-ctor(system-int32)).

Reply:
Yes, a nullable int would be a solution to what I mentioned.

Reply:
Decided on using `string` offline